lib/activefacts/generator/rails/active_admin.rb

```diff
@@ -0,0 +1,80 @@
+#
+#       ActiveFacts Rails Models Generator
+#
+# Copyright (c) 2018 Daniel Heath. Read the LICENSE file.
+#
+require 'digest/sha1'
+require 'activefacts/metamodel'
+require 'activefacts/compositions'
+require 'activefacts/generator'
+require 'activefacts/compositions/traits/rails'
+require 'activefacts/generator/rails/ruby_folder_generator'
+
+module ActiveFacts
+  module Generators
+    module Rails
+      class ActiveAdmin
+        include RubyFolderGenerator
+        def self.options
+          ({
+            keep:          ['Boolean', "Keep stale files"],
+            output:        [String,    "Write admin config files into this output directory"],
+          })
+        end
+
+        def self.compatibility
+          # REVISIT: We depend on the surrogate option being enabled if any PK is not Rails-friendly
+          [1, %i{relational}]   # one relational composition
+        end
+
+        def initialize constellation, composition, options = {}
+          @constellation = constellation
+          @composition = composition
+          @options = options
+          @option_keep = options.delete("keep")
+          @option_output = options.delete("output")
+          @option_output ||= 'app/admin'
+          @option_output = nil if @option_output == "-"
+        end
+
+        def generate_files
+          admins =
+            @composition.
+            all_composite.
+            sort_by{|composite| composite.mapping.name}.
+            map{|composite| generate_admin composite}.
```

Comment:
I would add a filter{needs_admin} here to remove any composites that have the "static" mapping pragma, indicating that this table should be loaded once at database build and not touched again.